lib/portage/dbapi/bintree.py

```diff
@@ -839,11 +839,18 @@ def _ensure_dir(self, path):
             return
         pkgdir_gid = pkgdir_st.st_gid
         pkgdir_grp_mode = 0o2070 & pkgdir_st.st_mode
-        try:
-            ensure_dirs(path, gid=pkgdir_gid, mode=pkgdir_grp_mode, mask=0)
-        except PortageException:
-            if not os.path.isdir(path):
-                raise
+
+        components = []
+        for component in path[len(self.pkgdir) + 1 :].split(os.sep):
```

Comment:
You might be able to do this more elegantly with pathlib.

Reply:
Yeah, relative_to and parts worked nicely here, thanks!
```python
for component in pathlib.PurePosixPath(path).relative_to(self.pkgdir).parts:
```

Reply:
Nice!